db/memtable.cc

```diff
@@ -516,9 +518,12 @@ bool MemTable::Add(SequenceNumber s, ValueType type,
                          std::memory_order_relaxed);
     }
 
-    if (prefix_bloom_) {
+    if (bloom_filter_) {
       assert(prefix_extractor_);
-      prefix_bloom_->Add(prefix_extractor_->Transform(key));
+      bloom_filter_->Add(prefix_extractor_->Transform(key));
```

Comment:
Can prefix_extractor_  be null if bloom_filter_ is not null?

Reply:
under current design, not possible:
https://github.com/facebook/rocksdb/blob/62acd648a598c6dcb0206d09a2ca06eb67aef704/db/memtable.cc#L114-L120
I'm actually wonder if we should have separate bloom filters for prefix and whole key modes because right now you cannot use whole key filter without configuring prefix_extractor, which seems a bit odd.

Reply:
We should allow whole key filtering without prefix extractor. But we still can do it with the same `bloom_filter_`, can we?

Reply:
sure they can share the same bloom filter, just to confirm: when both memtable_whole_key_filtering and prefix_extractor are set, the prefix bloom filter should be disabled, correct?